internal/internal_worker.go

```diff
@@ -1073,14 +1073,38 @@ func newAggregatedWorker(
 	}
 }
 
-func tagScope(metricsScope tally.Scope, tagName, tagValue string) tally.Scope {
+// tagScope with one or multiple tags, like
+// tagScope(scope, tag1, val1, tag2, val2)
+func tagScope(metricsScope tally.Scope, keyValueinPairs ...string) tally.Scope {
 	if metricsScope == nil {
 		metricsScope = tally.NoopScope
 	}
-	tagsMap := map[string]string{tagName: tagValue}
+	if len(keyValueinPairs)%2 != 0 {
+		panic("tagScope key value are not in pairs")
+	}
+	tagsMap := map[string]string{}
+	for i := 0; i < len(keyValueinPairs); i += 2 {
+		tagsMap[keyValueinPairs[i]] = keyValueinPairs[i+1]
+	}
 	return metricsScope.Tagged(tagsMap)
 }
 
+// getOrCreateTaggedScope is used to get or create single tag for metrics.
+// Note: This assume the map has no conflicts of tagValue for different tagName,
+// otherwise retrieved subScope will be corrupted. e.g you cannot getOrCreateTaggedScope(m, scope, "domain", "A name")
+// and then getOrCreateTaggedScope(m, scope, "workflow-type", "A name")
+func getOrCreateTaggedScope(m sync.Map, scope tally.Scope, tagName, tagValue string) tally.Scope {
+	taggedScope, ok := m.Load(tagValue)
```

Comment:
can you make the key be tagName+tagValue, then you don't have to worry about the case you put in your comment.